Source/WebCore/html/HTMLSummaryElement.cpp

```diff
@@ -101,7 +103,23 @@ static bool isClickableControl(EventTarget* target)
     if (!is<Element>(target))
         return false;
     auto& element = downcast<Element>(*target);
-    return is<HTMLFormControlElement>(element) || is<HTMLFormControlElement>(element.shadowHost());
```

Comment:
The old code checks the shadowHost too (and Chromium's code also seems to iterate over shadowHosts as well fwiw). Not sure this matters.

Reply:
> The old code checks the shadowHost too (and Chromium's code also seems to iterate over shadowHosts as well fwiw). Not sure this matters.

Yeah, I wondered about that. It’s not clear to me what the spec says about checking the shadowHost. It seems like another place where ideally we’d want to have some clarification added to the spec. If we get that clarification made to the spec — or if the spec is already clear on this, but I haven’t been looking in the right place — I’m happy to do a separate follow-up PR. But again this seems like another case where we may not have tests for it yet (in which case I’d need to write them, along with the WebKit patch).

Reply:
Sounds good, just wanted to make sure we're aware of the extent of the changes here :)

Reply:
> The old code checks the shadowHost too (and Chromium's code also seems to iterate over shadowHosts as well fwiw). Not sure this matters.

So it turns out that it does matter — for the cases of `textarea` and `input type=text` and I assume for other `input` types as well. (`textarea` and `input type=text` are the ones we have tests for.)

In those cases, when you click with them, the events get fired not at those elements themselves, but instead at `div` elements they host in the UA shadow DOM.